src/ClientResolver.php

```diff
@@ -118,6 +118,12 @@ class ClientResolver
             'doc'   => 'Allows you to specify which profile to use when credentials are created from the AWS credentials file in your HOME directory. This setting overrides the AWS_PROFILE environment variable. Note: Specifying "profile" will cause the "credentials" key to be ignored.',
             'fn'    => [__CLASS__, '_apply_profile'],
         ],
+        'config_profile' => [
```

Comment:
I'd like to see if we can limit the API surface area here and utilize a single option. I think it would be a good idea to see how all of the other SDKs are handling this.